File: daily-exercise/high_frequency/2sum.py
# -*- coding: utf-8 -*-
# @Time    : 2022/7/21 13:24
# @Author  : kevin
# @Version : python 3.7
# @Desc    : 2sum
# 哈希表存储每个数索引，查找target-当前数的差值
def two_sum(nums, target):
    hash_num_index = dict()
    for idx, nums in enumerate(nums):
        diff = target - nums
        if diff in hash_num_index:
            return [idx, hash_num_index[diff]]
        hash_num_index[nums] = idx

File: daily-exercise/high_frequency/2sum.go
// 哈希表存储每个数索引，查找target-当前数的差值
func twoSum(nums []int, target int) []int {
    hashNumIndex := make(map[int]int)
    for idx, num := range nums {
        diff := target - num
		if v, ok := hashNumIndex[diff]; ok {
			return []int{idx, v}
		}
        hashNumIndex[num] = idx
    }
    return nil
}